application-analytics-test/application-analytics-test-pageobjects/src/main/java/com/xwiki/analytics/test/po/HomePageEditPage.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package com.xwiki.analytics.test.po;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.interactions.Actions;
+import org.xwiki.test.ui.XWikiWebDriver;
+import org.xwiki.test.ui.po.editor.EditPage;
+
+public class HomePageEditPage extends EditPage
+{
+    private final XWikiWebDriver driver;
+
+    public HomePageEditPage()
+    {
+        driver = getUtil().getDriver();
+    }
+
+    /**
+     * Adds a new macro to the homepage of the application.
+     * @param macroID id of the macro
+     * @param macroName name of the macro
+     * @return
+     */
+
+    public HomePageEditPage addNewMacro(String macroID, String macroName)
+    {
+
+        this.clickAddGadget().selectMacro(macroID, macroName);
+        return this;
+    }
+
+    /**
+     * Removes the last macro that is on the page.
+     * @return
+     */
+
+    public HomePageEditPage removeLastMacro()
```

Comment:
This method seems fragile. Anyway, it should be part of the https://github.com/xwiki/xwiki-platform/blob/master/xwiki-platform-core/xwiki-platform-ckeditor/xwiki-platform-ckeditor-test/xwiki-platform-ckeditor-test-pageobjects/src/main/java/org/xwiki/ckeditor/test/po/MacroDialogSelectModal.java API. If you added it only for the edit permission check, I feel that it's safer to only test the add macro usecase. WDYT?